pre_commit_hooks/check_vcs_permalinks.py

```diff
@@ -7,8 +7,11 @@
 from typing import Sequence
 
 
+SHA_REGEX = re.compile(rb'^[a-fA-F0-9]{4,64}$')
+
+
 def _get_pattern(domain: str) -> Pattern[bytes]:
-    regex = rf'https://{domain}/[^/ ]+/[^/ ]+/blob/master/[^# ]+#L\d+'
+    regex = rf'https://{domain}/[^/ ]+/[^/ ]+/blob/([^/. ]+)/[^# ]+#L\d+'
```

Comment:
hmm I feel like this can be done with the pattern entirely (and not need to change the rest of the code) -- perhaps with lookaheads (atomic matches)?

Reply:
Yeah, you're right, I unnecessarily avoided the use of lookahead. I reverted the changes and updated just the regex this time.